.github/workflows/docs-gen-and-push.yml

```diff
@@ -30,7 +30,7 @@ jobs:
           echo "github.actor=${{ github.actor }}"
           echo "github.action_ref=${{ github.action_ref }}"
           echo "github.event_name=${{ github.event_name }}"
-          echo "github.head_ref=${{ github.head_ref }}"
+          echo "github.head_ref=$GITHUB_HEAD_REF"
```

Comment:
Oddly, both https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#github-context and https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#default-environment-variables say that these are only defined for PR events. But the old code is observed to work for workflow_dispatch too.